corehq/warehouse/transforms/sql/app_status_fact.sql

```diff
@@ -1,45 +1,37 @@
-SELECT * FROM
-(
+INSERT INTO {{ app_status_fact }} (
+       last_form_submission_date,
+       last_form_app_build_version,
+       last_form_app_commcare_version,
+       user_id,
+       app_id,
+       domain,
+       last_sync_log_date,
+       last_sync_log_app_build_version,
+       batch_id
+)
 
-    SELECT
-        max(last_submission) as last_submission,
-        max(last_submission_id) as last_submission_id,
-        user_id,
-        app_id
-    FROM (
-        SELECT
-            first_value(received_on AT TIME ZONE 'UTC') OVER wnd AS last_submission,
-            first_value(form_id) OVER wnd AS last_submission_id,
-            user_id,
-            app_id
-        FROM {{ form_staging }}
-        WINDOW wnd AS (
-            PARTITION BY user_id, app_id ORDER BY received_on DESC
-            ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
-        )
-    ) as t
-    GROUP BY user_id, app_id
-) as form_table
+SELECT 
+       form_table.last_submission,
+       form_table.submission_build_version,
+       form_table.commcare_version,
+       form_table.user_id,
+       form_table.app_id,
+       form_table.domain,
+       sync_table.last_sync,
+       sync_table.sync_build_version,
+       '{{ batch_id }}'
 
- LEFT OUTER JOIN
-
-(
-    SELECT
-        max(last_sync) as last_sync,
-        max(user_id) as user_id,
-        max(build_id) as build_id
-    FROM (
-        SELECT
-            first_value(sync_date AT TIME ZONE 'UTC') OVER wnd AS last_sync,
-            first_value(user_id) OVER wnd AS user_id,
-            first_value(build_id) OVER wnd AS build_id
-        FROM {{ synclog_staging }}
-        WINDOW wnd AS (
-            PARTITION BY user_id ORDER BY sync_date AT TIME ZONE 'UTC' DESC
-            ROWS BETWEEN UNBOUNDED PRECEDING AND UNBOUNDED FOLLOWING
-        )
-     ) as t
-     GROUP BY user_id
-) as sync_table
-
-ON sync_table.user_id = form_table.user_id
+FROM
+    {{ app_status_form_staging }} as form_table
+    LEFT JOIN {{ app_status_synclog_staging }} as sync_table
+    ON form_table.user_id = sync_table.user_id
```

Comment:
is the grain for both of these staging tables just one record per user?

Reply:
yep